sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample07_MockEventProcessorHandlersTests.cs

```diff
@@ -0,0 +1,244 @@
+                }
+            }
+
+            // Initialize the args and simulate checkpoint update method.
+
+            var partitionId = "0";
+            var eventBody = new BinaryData("This is a sample event body");
+            var eventData = EventHubsModelFactory.EventData(eventBody);
+            var partitionContext = EventHubsModelFactory.PartitionContext(partitionId);
+            var eventArgs = new ProcessEventArgs(partitionContext, eventData, _ => { checkpointUpdated = true; return Task.CompletedTask; });
+
+            for (int i = 1; i <= EventsBeforeCheckpoint; i++)
+            {
+                // Execute the handler and validate that the event handling has been checkpointed.
+
+                await processEventHandler(eventArgs);
+
+                if (i % EventsBeforeCheckpoint == 0)
+                    Assert.IsTrue(checkpointUpdated);
+                else
+                    Assert.IsFalse(checkpointUpdated);
+
+                Assert.That(partitionEventCount[partitionId], Is.EqualTo(i % EventsBeforeCheckpoint));
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ProcessErrorHandlerCancellationTrigger()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_ProcessErrorHandlerCancellationTrigger
+
+            using var cancellationSource = new CancellationTokenSource();
+
+            // Applicaition handler cancels events processing after detection of the
+            // exact error with exact operation.
+
+            Task processErrorHandler(ProcessErrorEventArgs args)
+            {
+                if (args.Exception.Message.Equals("Example exception") && args.Operation.Equals("Example operation"))
+                {
+                    cancellationSource.Cancel();
+                }
+
+                return Task.CompletedTask;
+            }
+
+            // Initialize the args with the specific operation name and exception message.
+
+            var exception = new Exception("Example exception");
+            var eventArgs = new ProcessErrorEventArgs("0", "Example operation", exception);
+
+            // Execute the handler and validate that the specific event arguments were detected
+            // and the cancellation has been triggered.
+
+            await processErrorHandler(eventArgs);
+            Assert.IsTrue(cancellationSource.IsCancellationRequested);
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic smoke test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task InitializeEventHandlerCancellationRequest()
+        {
+            #region Snippet:EventHubs_Processor_Sample07_InitializeEventHandlerCancellationRequest
+
+            // Application handler honors cancellation and doesn't set the
```

Comment:
👍   This comment fits perfectly for short and helpful!

Reply:
That's your comment... :)

Reply:
Well, that's both mildly embarrassing and yet, makes perfect sense that I'd like it, then.  😛